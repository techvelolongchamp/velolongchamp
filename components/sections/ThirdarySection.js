import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'

import Content from '../ui/Content'

import { SectionContainer, H1Title, H2Title } from './ThirdarySection.styled'

const ThirdarySection = forwardRef(
  ({ id, title, children, height, useH1, hideH1 }, ref) => (
    <SectionContainer name={id} id={id} height={height} ref={ref}>
      <Content>
        {useH1 ? (
          <H1Title hideH1={hideH1}>{title}</H1Title>
        ) : (
          <H2Title>{title}</H2Title>
        )}
        {children}
      </Content>
    </SectionContainer>
  )
)

ThirdarySection.propTypes = {
  id: PropTypes.string,
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  height: PropTypes.string,
  useH1: PropTypes.bool,
  hideH1: PropTypes.bool,
}

ThirdarySection.defaultProps = {
  useH1: false,
  hideH1: false,
}

export default ThirdarySection
