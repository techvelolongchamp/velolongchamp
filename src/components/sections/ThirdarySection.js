import React from 'react'
import styled from 'styled-components'

import Content from '../Content'

const SectionContainer = styled.section`
  height: ${({ theme, height }) => (height ? height : theme.sectionHeight)};
`

const Title = styled.h2`
  color: ${({ theme }) => theme.colors.defaultColor};
  font-size: ${({ theme }) => theme.fonts.xlarge};
  margin: ${({ theme }) => theme.spacing.l};
`

const ThirdarySection = ({ id, title, children, height }) => (
  <SectionContainer id={id} height={height}>
    <Content>
      <Title>{title}</Title>
      {children}
    </Content>
  </SectionContainer>
)

export default ThirdarySection
