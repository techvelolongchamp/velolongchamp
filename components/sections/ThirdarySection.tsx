import React, { forwardRef } from 'react'

import Content from '../ui/Content'

import { SectionContainer, H1Title, H2Title } from './ThirdarySection.styled'

const ThirdarySection = forwardRef<
  HTMLDivElement,
  {
    id?: string
    title: string
    height?: string
    useH1?: boolean
    hideH1?: boolean
    children: React.ReactNode
  }
>(({ id, title, children, height, useH1, hideH1 }, ref) => (
  <SectionContainer id={id} height={height} ref={ref}>
    <Content>
      {useH1 ? (
        <H1Title hideH1={hideH1}>{title}</H1Title>
      ) : (
        <H2Title>{title}</H2Title>
      )}
      {children}
    </Content>
  </SectionContainer>
))

ThirdarySection.displayName = 'ThirdarySection'

export default ThirdarySection
