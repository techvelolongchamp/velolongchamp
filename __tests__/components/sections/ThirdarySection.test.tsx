import React, { useRef } from 'react'
import { RenderWithProvider as render } from '../../provider'

import ThirdarySection from '../../../components/sections/ThirdarySection'

const MockComponent = (props) => {
  const ref = useRef(null)

  return <ThirdarySection ref={ref} {...props} />
}

describe('ThirdarySection unit tests', () => {
  it('should render correctly', () => {
    const { container } = render(<MockComponent>Hello</MockComponent>)

    expect(container).toMatchSnapshot()
  })

  it('should render correctly - with id, title and height - h2', () => {
    const { container, getByTestId } = render(
      <MockComponent height="300px" id="my-id" title="section-title">
        Hello
      </MockComponent>
    )
    const title = getByTestId('h2-title')
    expect(container).toMatchSnapshot()
    expect(title).toBeInTheDocument()
  })

  it('should render correctly - with id, title and height - h1', () => {
    const { container, getByTestId } = render(
      <MockComponent height="300px" id="my-id" title="section-title" useH1>
        Hello
      </MockComponent>
    )
    const title = getByTestId('h1-title')
    expect(container).toMatchSnapshot()
    expect(title).toBeInTheDocument()
  })

  it('should render correctly - with id, title and height - h1 hidden', () => {
    const { container, getByTestId } = render(
      <MockComponent
        height="300px"
        id="my-id"
        title="section-title"
        useH1
        hideH1
      >
        Hello
      </MockComponent>
    )
    const title = getByTestId('h1-title')
    expect(container).toMatchSnapshot()
    expect(title).toBeInTheDocument()
  })
})
