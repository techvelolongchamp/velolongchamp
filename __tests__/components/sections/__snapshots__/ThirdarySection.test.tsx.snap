// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThirdarySection unit tests should render correctly - with id, title and height - h1 1`] = `
.c1 {
  margin: 0 auto;
  max-width: 750px;
  padding: 10px;
  height: inherit;
}

.c0 {
  min-height: 300px;
}

.c2 {
  color: #419a1c;
  font-size: 30px;
  margin: 20px 0;
}

<div>
  <section
    class="c0"
    height="300px"
    id="my-id"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
        data-testid="h1-title"
      >
        section-title
      </h1>
      Hello
    </div>
  </section>
</div>
`;

exports[`ThirdarySection unit tests should render correctly - with id, title and height - h1 hidden 1`] = `
.c1 {
  margin: 0 auto;
  max-width: 750px;
  padding: 10px;
  height: inherit;
}

.c0 {
  min-height: 300px;
}

.c2 {
  display: none;
  color: #419a1c;
  font-size: 30px;
  margin: 20px 0;
}

<div>
  <section
    class="c0"
    height="300px"
    id="my-id"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
        data-testid="h1-title"
      >
        section-title
      </h1>
      Hello
    </div>
  </section>
</div>
`;

exports[`ThirdarySection unit tests should render correctly - with id, title and height - h2 1`] = `
.c1 {
  margin: 0 auto;
  max-width: 750px;
  padding: 10px;
  height: inherit;
}

.c0 {
  min-height: 300px;
}

.c2 {
  color: #419a1c;
  font-size: 30px;
  margin: 20px 0;
}

<div>
  <section
    class="c0"
    height="300px"
    id="my-id"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
        data-testid="h2-title"
      >
        section-title
      </h2>
      Hello
    </div>
  </section>
</div>
`;

exports[`ThirdarySection unit tests should render correctly 1`] = `
.c1 {
  margin: 0 auto;
  max-width: 750px;
  padding: 10px;
  height: inherit;
}

.c0 {
  min-height: 600px;
}

.c2 {
  color: #419a1c;
  font-size: 30px;
  margin: 20px 0;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
        data-testid="h2-title"
      />
      Hello
    </div>
  </section>
</div>
`;
